tasks/rh-sign-image-cosign/rh-sign-image-cosign.yaml

```
@@ -71,8 +77,16 @@ spec:
             if [ "$MEDIA_TYPE" = "application/vnd.docker.distribution.manifest.list.v2+json" ]; then LIST=1; fi
             if [ "$MEDIA_TYPE" = "application/vnd.oci.image.index.v1+json" ]; then LIST=1; fi
 
+            # Upload transparency log when rekor url is specified
+            set +u
+            if [ -z "$REKOR_URL" ]; then
+                COSIGN_COMMON_ARGS="--tlog-upload=false --key $SIGN_KEY"
+            else
+                COSIGN_COMMON_ARGS="-y --rekor-url=$REKOR_URL --key $SIGN_KEY"
+            fi
+            set -u
```

You could do `if [ -v REKOR_URL ]` and revert the logic and then `set +u` would not be needed. WDYT?
```suggestion
            if [ -v REKOR_URL ]; then
                COSIGN_COMMON_ARGS="-y --rekor-url=$REKOR_URL --key $SIGN_KEY"
            else
                COSIGN_COMMON_ARGS="--tlog-upload=false --key $SIGN_KEY"
            fi
```

Unless you're worried that it could be set, but empty.

---

Good point. It should be either set with non-empty value or not set. The suggestion is applied.